akka-cluster/src/main/scala/akka/cluster/ClusterHeartbeat.scala

```diff
@@ -116,14 +121,19 @@ private[cluster] final class ClusterHeartbeatSender extends Actor with ActorLogg
   }
 
   def init(snapshot: CurrentClusterState): Unit = {
-    val nodes: Set[UniqueAddress] = snapshot.members.map(_.uniqueAddress)
-    val unreachable: Set[UniqueAddress] = snapshot.unreachable.map(_.uniqueAddress)
+    val nodes = snapshot.members.filter(filterinternalClusterMembers).map(_.uniqueAddress)
+    val unreachable = snapshot.unreachable.filter(filterinternalClusterMembers).map(_.uniqueAddress)
     state = state.init(nodes, unreachable)
   }
 
   def addMember(m: Member): Unit =
-    if (m.uniqueAddress != selfUniqueAddress && !state.contains(m.uniqueAddress))
+    if (m.uniqueAddress != selfUniqueAddress && // is not self 
+      !state.contains(m.uniqueAddress) && // not already added
+      filterinternalClusterMembers(m) // should be watching members from this team (internal / external)
+      ) {
+      log.info(s"WILL MONITOR NODE ===== ${m.uniqueAddress} ======== TEAM: ${m.team} (myself team: ${cluster.selfDatacenter})")
       state = state.addMember(m.uniqueAddress)
+    }
 
   def removeMember(m: Member): Unit =
```

Comment:
add the filter condition here also, shouldn't be needed but might save some work done by `state.removeMember`